### PR TITLE
Feature/sdk implementation

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,63 +1,41 @@
...
-  "name": "iexec-data-protector",
-  "version": "1.0.0",
-  "description": "Securely protect your data by creating a confidential NFT (cNFT). You'll have complete control over access to your data, keeping it private and managing its use by other applications.",
-  "keywords": [
-    "iExec",
-    "NFT",
-    "cNFT"
-  ],
-  "main": "src/index.tsx",
+  "name": "vite-project",
```
Comment (SeddikBellamine) [inline package.json:2]: are we sure about the name of the sandbox ?

File: package.json [hunk 2]
```diff
@@ -1,63 +1,41 @@
...
+  "private": true,
```
Comment (SeddikBellamine) [inline package.json:3]: i think we should delete the private property.

File: src/features/grantAccess/GrantAccess.tsx [hunk 3]
```diff
@@ -102,21 +100,21 @@ export default function GrantAccess() {
...
+        error={!isValidAuthorizedUser}
         helperText={
-          !isValidUserRestrictedAddress && 'Please enter a valid user address'
+          !isValidAuthorizedUser && "Please enter a valid user address"
         }
       />
       {!loading && (
         <Button
-          sx={{ display: 'block', margin: '20px auto' }}
+          sx={{ display: "block", margin: "20px auto" }}
```
Comment (SeddikBellamine) [inline src/features/grantAccess/GrantAccess.tsx]: may be put styling in a seperate css file ?

File: src/features/grantAccess/GrantAccess.tsx [hunk 4]
```diff
@@ -139,9 +137,9 @@ export default function GrantAccess() {
       )}
       {loading && (
         <CircularProgress
-          sx={{ display: 'block', margin: '20px auto' }}
+          sx={{ display: "block", margin: "20px auto" }}
```
Comment (SeddikBellamine) [inline src/features/grantAccess/GrantAccess.tsx]: may be put styling in a seperate css file ?

File: src/features/revokeAccess/RevokeAccess.tsx [hunk 5]
```diff
@@ -5,50 +5,50 @@ import {
...
+import { useState } from "react";
+import revokeAccessFunc from "./revokeAccessFunc";
+import { isAddress } from "ethers/lib/utils.js";
+import { useAppSelector } from "../../app/hooks";
 import {
   selectProtectedDataCreated,
   selectUserAddressRestricted,
-} from '../../app/appSlice'
+} from "../../app/appSlice";
+import { NULL_ADDRESS } from "../../utils/constant";
```
Comment (SeddikBellamine) [inline src/features/revokeAccess/RevokeAccess.tsx:17]: you can use the NULL_ADDRESS in iexec utils: 
`import { utils } from 'iexec';`

https://github.com/iExecBlockchainComputing/iexec-sdk/blob/master/docs/modules/utils.md

File: src/features/revokeAccess/RevokeAccess.tsx [hunk 6]
```diff
@@ -5,50 +5,50 @@ import {
...
     } catch (error) {
-      setError(String(error))
-      setRevokeAccess([''])
+      setError(String(error));
+      setRevokeAccess("");
     }
-    console.log(dataProtectedAddress)
-    setLoading(false)
-  }
+    console.log(protectedData);
```
Comment (SeddikBellamine) [inline src/features/revokeAccess/RevokeAccess.tsx]: delete console.log

File: src/features/revokeAccess/RevokeAccess.tsx [hunk 7]
```diff
@@ -58,17 +58,17 @@ export default function RevokeAccess() {
...
+        error={!isValidProtectedData}
         helperText={
-          !isValidDataProtectedAddress && 'Please enter a valid dataset address'
+          !isValidProtectedData && "Please enter a valid protectedData address"
         }
       />
       {!loading && (
         <Button
-          sx={{ display: 'block', margin: '20px auto' }}
+          sx={{ display: "block", margin: "20px auto" }}
```
Comment (SeddikBellamine) [inline src/features/revokeAccess/RevokeAccess.tsx:70]: styles in a css file

File: src/utils/constant.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+export const NULL_ADDRESS = '0x0000000000000000000000000000000000000000';
```
Comment (SeddikBellamine) [inline src/utils/constant.ts]: we can remove this file and use iexec/utils

File: package.json (hunk 1 above)
Comment (Le-Caignec) [reply package.json]: Commit: fc01c2b3f9e281912c314fdc603c0c2fe37a71f0

File: package.json (hunk 2 above)
Comment (Le-Caignec) [reply package.json]: Commit: fc01c2b3f9e281912c314fdc603c0c2fe37a71f0

File: src/features/revokeAccess/RevokeAccess.tsx (hunk 5 above)
Comment (Le-Caignec) [reply src/features/revokeAccess/RevokeAccess.tsx]: Commit: 2615df5d14a47a363e351642df82c96097e82bd5

Comment (Le-Caignec) [reply src/features/revokeAccess/RevokeAccess.tsx]: Impossible d'import `utils` depuis la lib `iexec`. Sans doute à cause de la configuration webpack de `dataprotector`
J'ai crée un ticket sur Jira : https://iexecproject.atlassian.net/browse/PRO-143?atlOrigin=eyJpIjoiZGU2YjVhODkwNjI4NDAzMWIyMTk3MWZiNjNkNWYwMDkiLCJwIjoiaiJ9

File: src/features/revokeAccess/RevokeAccess.tsx (hunk 6 above)
Comment (Le-Caignec) [reply src/features/revokeAccess/RevokeAccess.tsx]: done

File: src/utils/constant.ts (hunk 8 above)
Comment (Le-Caignec) [reply src/utils/constant.ts]: Impossible d'import `utils` depuis la lib `iexec`. Sans doute à cause de la configuration webpack de `dataprotector`
J'ai crée un ticket sur Jira : https://iexecproject.atlassian.net/browse/PRO-143?atlOrigin=eyJpIjoiZGU2YjVhODkwNjI4NDAzMWIyMTk3MWZiNjNkNWYwMDkiLCJwIjoiaiJ9

File: src/features/grantAccess/GrantAccess.tsx (hunk 4 above)
Comment (Le-Caignec) [reply src/features/grantAccess/GrantAccess.tsx]: Commit: 6ed78be006aa665d665e6c58c81ce34c9e155aab

File: src/features/grantAccess/GrantAccess.tsx (hunk 3 above)
Comment (Le-Caignec) [reply src/features/grantAccess/GrantAccess.tsx]: Commit: 6ed78be006aa665d665e6c58c81ce34c9e155aab

File: src/features/revokeAccess/RevokeAccess.tsx (hunk 7 above)
Comment (Le-Caignec) [reply src/features/revokeAccess/RevokeAccess.tsx]: Commit: 6ed78be006aa665d665e6c58c81ce34c9e155aab

**Review by SeddikBellamine — APPROVED** [review]
LGTM !

**Pull request merged**